src/main/java/mate/academy/spring/controller/OrderController.java

```diff
@@ -0,0 +1,52 @@
+package mate.academy.spring.controller;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import mate.academy.spring.mapper.impl.response.OrderResponseMapper;
+import mate.academy.spring.model.Order;
+import mate.academy.spring.model.ShoppingCart;
+import mate.academy.spring.model.User;
+import mate.academy.spring.model.dto.response.OrderResponseDto;
+import mate.academy.spring.service.OrderService;
+import mate.academy.spring.service.ShoppingCartService;
+import mate.academy.spring.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+    private final OrderService orderService;
+    private final ShoppingCartService shoppingCartService;
+    private final UserService userService;
+    private final OrderResponseMapper orderResponseMapper;
+
+    public OrderController(OrderService orderService,
+                           ShoppingCartService shoppingCartService,
+                           UserService userService,
+                           OrderResponseMapper orderResponseMapper) {
+        this.orderService = orderService;
+        this.shoppingCartService = shoppingCartService;
+        this.userService = userService;
+        this.orderResponseMapper = orderResponseMapper;
+    }
+
+    @PostMapping("/complete")
+    public OrderResponseDto completeByUserId(@RequestParam Long userId) {
+        User user = userService.get(userId);
+        ShoppingCart shoppingCart = shoppingCartService.getByUser(user);
+        Order order = orderService.completeOrder(shoppingCart);
+        return orderResponseMapper.toDto(order);
+    }
+
+    @GetMapping
+    public List<OrderResponseDto> getOrderHistoryByUserId(@RequestParam Long userId) {
```

Comment:
```suggestion
    public List<OrderResponseDto> getOrdersHistory(@RequestParam Long userId) {
```
keep to consistency with service layer method naming